### PR TITLE
Validate `CLOUDINARY_URL` scheme

### PR DESCRIPTION
**Pull request by leptians**
Validate `CLOUDINARY_URL` scheme

### CODE REVIEW

File: src/Cloudinary.php [hunk 1]
```diff
@@ -145,9 +145,15 @@ public static function config_from_url($cloudinary_url)
         if ($cloudinary_url) {
             $uri = parse_url($cloudinary_url);
             $q_params = array();
+
```
Comment (const-cloudinary) [inline src/Cloudinary.php]: This block can be moved above line 147, no point of allocating variable `$q_params` that won't be used in case exception is thrown.

File: src/Cloudinary.php [hunk 2]
```diff
@@ -145,9 +145,15 @@ public static function config_from_url($cloudinary_url)
...
+            if (strtolower($uri["scheme"]) !== "cloudinary") {
```
Comment (const-cloudinary) [inline src/Cloudinary.php]: `$uri` is not guaranteed to have a `scheme` key, if passing for example `" "` as `$clouidnary_url`, a notice: `Undefined index: scheme` is thrown. Please check that `$uri["scheme"]` is set. ( As an alternative, you can use `Cloudinary::option_get` method that does that for you)

File: tests/CloudinaryTest.php [hunk 3]
```diff
@@ -71,6 +71,17 @@ public function tearDown()
         \Cloudinary::$USER_PLATFORM = $this->original_user_platform;
     }
 
+    public function test_cloudinary_url()
```
Comment (const-cloudinary) [inline tests/CloudinaryTest.php]: `ConfigTest.php` seems to be more appropriate place to put this unit test.

File: tests/CloudinaryTest.php [hunk 4]
```diff
@@ -71,6 +71,17 @@ public function tearDown()
...
+    {
+        \Cloudinary::config_from_url("cloudinary://123456789012345:ALKJdjklLJAjhkKJ45hBK92baj3@test/test.example.com?cname=test.example.com");
```
Comment (const-cloudinary) [inline tests/CloudinaryTest.php]: Please do not exceed 120 characters in a single line (this one has 142 characters)

File: tests/CloudinaryTest.php [hunk 5]
```diff
@@ -71,6 +71,17 @@ public function tearDown()
...
+
+        try {
+            \Cloudinary::config_from_url("https://123456789012345:ALKJdjklLJAjhkKJ45hBK92baj3@test/test.example.com?cname=test.example.com");
```
Comment (const-cloudinary) [inline tests/CloudinaryTest.php]: Please add a few more failing tests, with, for example, empty scheme, or completely invalid cloudinary url ( string with a single character or some random string, etc) to make sure the code you wrote provides meaningful error messages and does not crash.

File: tests/CloudinaryTest.php [hunk 6]
```diff
@@ -71,6 +71,17 @@ public function tearDown()
         \Cloudinary::$USER_PLATFORM = $this->original_user_platform;
     }
 
+    public function test_cloudinary_url()
+    {
+        \Cloudinary::config_from_url("cloudinary://123456789012345:ALKJdjklLJAjhkKJ45hBK92baj3@test/test.example.com?cname=test.example.com");
+
```
Comment (const-cloudinary) [inline tests/CloudinaryTest.php]: this unit test can be separated into 2 tests, one for good schemes, another one for invalid schemes
